pydata_sphinx_theme/__init__.py

```diff
@@ -107,7 +107,9 @@ def add_header_level_recursive(ul, level):
                 if ul:
                     add_header_level_recursive(ul, level + 1)
 
-        add_header_level_recursive(soup.find("ul"), 1)
+        root_ul = soup.find("ul")
+        if root_ul:
+            add_header_level_recursive(root_ul, 1)
```

Comment:
an alternative might be to replace both the newly-added assign/if, and collapse the assign/if on 106/107, e.g. 

```py
        # Add toc-hN + visible classes
        def add_header_level_recursive(ul, level):
            if ul is not None:
                if level <= (context["theme_show_toc_level"] + 1):
                    ul["class"] = ul.get("class", []) + ["visible"]
                for li in ul("li", recursive=False):
                    li["class"] = li.get("class", []) + [f"toc-h{level}"]
                    add_header_level_recursive(li.find("ul", recursive=False), level + 1)

        add_header_level_recursive(soup.find("ul"), 1)
```

it's probably not a good idea for us to have been re-assigning `ul` while iterating over `ul`, or?

Reply:
I agree. Would you be ok with inverting the proposed `if ul is not None` to `if ul is None` and then return from the function immediately to reduce the nesting?